pkg/controller/daemon/daemoncontroller.go

```diff
@@ -994,32 +990,37 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 
 	if ds.DeletionTimestamp != nil || !dsc.expectations.SatisfiedExpectations(dsKey) {
 		// Only update status.
-		return dsc.updateDaemonSetStatus(ds, hash)
+		return dsc.updateDaemonSetStatus(ds, hash, nodeList)
 	}
 
-	err = dsc.manage(ds, hash)
+	err = dsc.manage(ds, hash, nodeList)
 	if err != nil {
 		return err
+	} else {
```

Comment:
If we do not enqueue this means that basically node relisting is not a probem, and we can factor out nodeList directly ? Or I'm I missing something ? @lukaszo @janetkuo